### PR TITLE
ETS-40 Add transfer submitter role

### PR DESCRIPTION
**Pull request by jazairi**
#### Why these changes are being introduced:

The Ability model needs a submitter role to control who has
permission to deposit and manage transfers.

#### Relevant ticket(s):

https://mitlibraries.atlassian.net/browse/ETS-40

#### How this addresses that need:

This adds the transfer submitter role, along with the preliminary
TransferController with create and show methods. (Additional logic
will be added to the controller as needed.)

The Ability model now includes the following additional permissions:
* thesis_admin can create transfers and view any transfer
* transfer_submitter can create transfers and view their own transfers
* processor can view any transfer

#### Side effects of this change:

* Generated the transfer dashboard and created a couple of stubbed
transfer views.
* Minor changes to our test fixtures to make the controller tests
clearer.
* Updated a thesis controller test assertion to redirect to the
login URL specifically, instead of redirecting anywhere.

#### Todo:
- [x] Tests
- [x] Documentation
- [x] Stakeholder approval

#### Requires Database Migrations?
NO

#### Includes new or updated dependencies?
NO

### CODE REVIEW

**Comment by jazairi** [comment]
Looks like the decrease in coverage is due to the new transfer dashboard. My assumption is that this wouldn't require coverage since it doesn't do anything yet, but I could be mistaken about that.

File: test/controllers/thesis_controller_test.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ class ThesisControllerTest < ActionDispatch::IntegrationTest
   # ~~~~~~~~~~~~~~~~~~~~~~~~~~ the submission system ~~~~~~~~~~~~~~~~~~~~~~~~~~
   test 'new prompts for login' do
     get '/thesis/new'
-    assert_response :redirect
+    assert_redirected_to '/users/auth/saml'
```
Comment (matt-bernhardt) [inline test/controllers/thesis_controller_test.rb:7]: Is this change related to anything else in the PR? I see that this matches what the test is for the transfer controller - so I'm reading this as "what the test should have been" rather than a substantive change. I'm fine with that, but don't want to miss anything.

Comment (jazairi) [reply test/controllers/thesis_controller_test.rb:7]: Yes, this change is unrelated to the goal of the PR. I noticed that the test doesn't really do what it seeks to, so I figured I'd fix it as I was writing similar tests for the TransferController.

**Review by matt-bernhardt — APPROVED** [review]
This all seems to be fine to me. The tests pass, and there's no obvious edge cases missed. Some of this will get its UI started in the next ticket, while other aspects may come a bit further down the road.

:shipit:
